datafusion/src/physical_plan/sorts/mod.rs

```diff
@@ -32,11 +33,10 @@ use std::borrow::BorrowMut;
 use std::cmp::Ordering;
 use std::fmt::{Debug, Formatter};
 use std::pin::Pin;
+use std::sync::atomic::AtomicUsize;
```

Comment:
for anyone else following along, the switch to using `AtomicUsize` I think is driven by needing to share the SortKeyCursors in a `BinaryHeap` as well as in a `Vec` of cursors (see comment below)